File: backend/app/api/audit_logs.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from app.core.permissions import require_permission
from app.dao.audit_log_dao import AuditLogDAO
from app.db.session import get_db
from app.models.audit_log import AuditLog
from app.models.user import User
from app.schemas.audit_log import AuditLogListResponse
from app.services.audit_log_service import list_audit_logs

router = APIRouter(prefix="/audit-logs", tags=["audit-logs"])


@router.get("", response_model=AuditLogListResponse)
def get_audit_logs(page: int = Query(1, ge=1), page_size: int = Query(20, ge=1, le=100), action: str | None = Query(default=None, max_length=100), actor_user_id: int | None = Query(default=None, ge=1), target_type: str | None = Query(default=None, max_length=50), target_id: str | None = Query(default=None, max_length=100), current_user: User = Depends(require_permission("audit_logs", "read")), db: Session = Depends(get_db)):
    filters = []
    if action:
        filters.append(AuditLog.action == action)
    if actor_user_id is not None:
        filters.append(AuditLog.actor_user_id == actor_user_id)
    if target_type:
        filters.append(AuditLog.target_type == target_type)
    if target_id:
        filters.append(AuditLog.target_id == target_id)
    items, total = list_audit_logs(db, page, page_size, filters)
    return {"items": items, "total": total, "page": page, "page_size": page_size}

File: frontend/src/api/auditLogs.ts
import { apiRequest } from './http'
import type { AuditLogListResponse } from '../types/auditLog'

export interface AuditLogQuery {
  page?: number
  page_size?: number
  action?: string
  actor_user_id?: number
  target_type?: string
  target_id?: string
}

export function listAuditLogs(
  params: AuditLogQuery = {},
): Promise<AuditLogListResponse> {
  const query = new URLSearchParams()
  query.set('page', String(params.page ?? 1))
  query.set('page_size', String(params.page_size ?? 20))

  if (params.action?.trim()) query.set('action', params.action.trim())
  if (params.actor_user_id) query.set('actor_user_id', String(params.actor_user_id))
  if (params.target_type?.trim()) query.set('target_type', params.target_type.trim())
  if (params.target_id?.trim()) query.set('target_id', params.target_id.trim())

  return apiRequest<AuditLogListResponse>(`/audit-logs?${query.toString()}`)
}
